src/drivers/heater/heater.cpp

```diff
@@ -0,0 +1,447 @@
+		work_queue(LPWORK, &_work, (worker_t)&Heater::cycle_trampoline, this,
+			   USEC2TICK(_controller_period_usec - _controller_time_on_usec));
+	}
+}
+
+void Heater::cycle_trampoline(void *arg)
+{
+	Heater *obj = reinterpret_cast<Heater *>(arg);
+	obj->cycle();
+}
+
+float Heater::duty_cycle()
+{
+	PX4_INFO("Average duty cycle:  %3.1f%%", (double)(_duty_cycle * 100.f));
+	return _duty_cycle;
+}
+
+float Heater::feed_forward(char *argv[])
+{
+	if (argv[1]) {
+		_p_feed_forward_value.set(atof(argv[1]));
+
+	}
+
+	PX4_INFO("feedforward value:  %2.5f", (double)_p_feed_forward_value.get());
+	return _p_feed_forward_value.get();
+}
+
+void Heater::initialize_topics()
+{
+	// Get the total number of accelerometer instances.
+	size_t number_of_imus = orb_group_count(ORB_ID(sensor_accel));
+
+	// Check each instance for the correct ID.
+	for (size_t x = 0; x < number_of_imus; x++) {
+		_sensor_accel_subscription = orb_subscribe_multi(ORB_ID(sensor_accel), (int)x);
+
+		while (orb_update(ORB_ID(sensor_accel), _sensor_accel_subscription, &_sensor_accel) != true) {
+			usleep(200000);
+		}
+
+		// If the correct ID is found, exit the for-loop with _sensor_accel_subscription pointing to the correct instance.
+		if (_sensor_accel.device_id == (uint32_t)_p_sensor_id.get()) {
+			PX4_INFO("Found Sensor to Temp Compensate");
+			break;
+		}
+	}
+
+	PX4_INFO("Device ID:  %d", _sensor_accel.device_id);
+
+	// Exit the driver if the sensor ID does not match the desired sensor.
+	if (_sensor_accel.device_id != (uint32_t)_p_sensor_id.get()) {
+		request_stop();
+		PX4_INFO("Could not find sensor to control temperature");
+	}
+}
+
+void Heater::initialize_trampoline(void *arg)
+{
+	Heater *heater = new Heater();
+
+	if (!heater) {
+		PX4_ERR("Heater driver alloc failed");
+		return;
+	}
+
+	heater->start();
+	_object = heater;
+}
+
+float Heater::integrator(char *argv[])
+{
+	if (argv[1]) {
+		_p_integrator_gain.set(atof(argv[1]));
+	}
+
+	PX4_INFO("integrator gain:  %2.5f", (double)_p_integrator_gain.get());
+	return _p_integrator_gain.get();
+}
+
+bool Heater::orb_update(const struct orb_metadata *meta, int handle, void *buffer)
+{
+	bool newData = false;
+
+	// Check if there is new data to obtain.
+	if (orb_check(handle, &newData) != OK) {
+		return false;
+	}
+
+	if (!newData) {
+		return false;
+	}
+
+	if (orb_copy(meta, handle, buffer) != OK) {
+		return false;
+	}
+
+	return true;
+}
+
+int Heater::print_status()
+{
+	PX4_INFO("Temp: %3.3f - Target Temp: %3.2f - Heater State: %s",
+		 (double)_sensor_temperature,
+		 (double)_p_temperature_setpoint.get(),
+		 _heater_on ? "On" : "Off");
+
+	return PX4_OK;
+}
+
+int Heater::print_usage(const char *reason)
+{
+	if (reason) {
+		printf("%s\n\n", reason);
+	}
+
+	PRINT_MODULE_DESCRIPTION(
+		R"DESCR_STR(
+### Description
+Background process running periodically on the LP work queue to regulate IMU temperature at a setpoint.
+
+The tasks can be started via CLI or uORB topics (vehicle_command from MAVLink, etc.).
```

Comment:
`or uORB topics (vehicle_command from MAVLink, etc.)` I don't see this?

Reply:
Fixed.  (Copy/pasta from send_event).  Thanks!